extensions/vscode/src/views/homeView.ts

```diff
@@ -1304,32 +1387,41 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
         entrypoint,
       );
       if (selected) {
-        window.showInformationMessage(
-          `A new deployment has been created for ${entrypoint}. Make any changes needed and
-					then click 'Deploy Your Project'to publish it`,
-        );
+        // publish!
+        return this.initiatePublish(selected.publishParams);
       }
-      return;
+      return false;
     }
-    // only one deployment, just activate it
+    // only one deployment, just publish it
     if (compatibleContentRecords.length === 1) {
       const contentRecord = compatibleContentRecords[0];
       const deploymentSelector: DeploymentSelector = {
         deploymentPath: contentRecord.deploymentPath,
       };
-      this.propagateDeploymentSelection(deploymentSelector);
-      window.showInformationMessage(
-        `An existing deployment for ${entrypoint} has been selected. After making any changes 
-					needed to your deployment configuration and your project settings,
-					you can click 'Deploy Your Project'to publish it.`,
-      );
+      await this.propagateDeploymentSelection(deploymentSelector);
+      const credentialName =
+        this.getCredentialNameForContentRecord(contentRecord);
+      if (!credentialName) {
+        window.showErrorMessage(
+          `Error: Unable to Deploy. No credential found for server at ${contentRecord.serverUrl}`,
+        );
+        return;
+      }
 
-      return deploymentSelector;
+      const target: PublishProcessParams = {
+        deploymentName: contentRecord.saveName,
+        credentialName,
+        configurationName: contentRecord.configurationName,
+        projectDir: contentRecord.projectDir,
+      };
+      // publish!
+      return this.initiatePublish(target);
     }
     // if there are multiple compatible deployments, then make sure one of these isn't
     // already selected. If it is, do nothing, otherwise pick between the compatible ones.
     const currentContentRecord = this.getActiveContentRecord();
     if (
+      !currentContentRecord ||
```

Comment:
Oh weird not sure why this comment got lost. I was noting that this felt strange to me. Is this not already caught by the `find()` logic? I just want to be sure I'm not missing something in my understanding.